src/workspacer/Workspace/WorkspaceManager.cs

```diff
@@ -28,6 +28,73 @@ public WorkspaceManager(IConfigContext context)
             _context = context;
             _windowsToWorkspaces = new Dictionary<IWindow, IWorkspace>();
         }
+        
+        public void FocusNextWindow()
+        {
+            var windows = FocusedWorkspace.ManagedWindows;
+            var didFocus = false;
+            for (var i = 0; i < windows.Count; i++)
+            {
+                var window = windows[i];
+                if (window.IsFocused)
+                {
+                    if (i + 1 == windows.Count)
+                    {
+                        //go to next monitor that has windows
+                        do 
+                        {
+                            this.SwitchFocusToNextMonitor();
+                        }
+                        while(!FocusedWorkspace.ManagedWindows.Any());
```

Comment:
In theory, if you have no windows open and trigger this method, won't this code loop forever?

I think we may benefit by extracting this code into its own function (`SwitchFocusToNextMonitorWithWindows()` ?) where we can implement a better check without complicating this method.

Reply:
yes I didn’t see that. Will update